php/libraries/LorisForm.class.inc

```diff
@@ -373,7 +373,7 @@ class LorisForm
                 $optionKey = (string) $optionKey;
             }
 
-            if ($optionKey === $this->getValue($el['name'])) {
+            if ($optionKey === $el['selected']) {
```

Comment:
After talking to you: I think the two cases are covering different cases (option pre-selected in the form vs user selected something and hit submit), this should probably be an "or" so that both cases are covered.